src/rust/engine/process_execution/src/named_caches.rs

```diff
@@ -20,7 +20,22 @@ impl NamedCache {
   }
 }
 
-const WORKSPACE_CACHE_BASE: &str = ".cache";
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+pub struct CacheDest(String);
+
+impl CacheDest {
+  pub fn new(dest: String) -> Result<CacheDest, String> {
+    let dest_path = PathBuf::from(&dest);
+    if dest_path.is_relative() && dest_path.components().next().is_some() {
```

Comment:
This is not a robust check. You should probably use RelativePath: https://github.com/pantsbuild/pants/blob/8621b189b788fa993c7fad1016091c3aedd385f0/src/rust/engine/process_execution/src/lib.rs#L155-L173